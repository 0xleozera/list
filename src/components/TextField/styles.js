import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: "row",
    marginTop: 10,
    marginBottom: 10
  },
  icon: {
    padding: 10,
    borderWidth: 1,
    borderColor: "#FFF",
    borderTopLeftRadius: 8,
    borderBottomLeftRadius: 8,
    marginLeft: 10,
    height: 40,
    backgroundColor: "#FFF"
  },
  field: {
    flex: 1,
    paddingTop: 10,
    paddingRight: 10,
    paddingBottom: 10,
    paddingLeft: 0,
    borderWidth: 1,
    borderColor: "#FFF",
    borderTopRightRadius: 8,
    borderBottomRightRadius: 8,
    marginRight: 10,
    height: 40,
    backgroundColor: "#FFF"
  }
});

export default styles;
