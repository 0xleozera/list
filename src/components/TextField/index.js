import React, { Component } from "react";
import { View, TextInput, Platform } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";

import styles from "./styles";

class TextField extends Component {
  getIcon() {
    return Platform.OS === "ios" ? "ios-search" : "md-search";
  }

  render() {
    return (
      <View style={styles.wrapper}>
        <Icon
          style={styles.icon}
          name={this.getIcon()}
          size={20}
          color="#000"
        />
        <TextInput
          style={styles.field}
          onChangeText={searchString => this.setState({ searchString })}
          underlineColorAndroid="transparent"
        />
      </View>
    );
  }
}

export default TextField;
