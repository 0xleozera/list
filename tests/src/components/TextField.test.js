import React from "react";
import { create } from "react-test-renderer";
import TextField from "../../../src/components/TextField";

describe("[Component] TextField", () => {
  const props = {
    onChange: jest.fn(),
    value: "",
    icon: ""
  };

  const tree = create(<TextField {...props} />);

  it("matches snapshot", () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("calls onChange prop", () => {
    tree.root.props.onChange();
    expect(props.onChange).toHaveBeenCalled();
  });
});
