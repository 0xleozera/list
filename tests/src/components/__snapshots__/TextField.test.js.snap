// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] TextField matches snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#8F8F91",
          "fontSize": 20,
        },
        Object {
          "backgroundColor": "#FFF",
          "borderBottomLeftRadius": 8,
          "borderColor": "#FFF",
          "borderTopLeftRadius": 8,
          "borderWidth": 1,
          "height": 40,
          "marginLeft": 10,
          "padding": 10,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï’¥
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderBottomRightRadius": 8,
        "borderColor": "#FFF",
        "borderTopRightRadius": 8,
        "borderWidth": 1,
        "flex": 1,
        "height": 40,
        "marginRight": 10,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
